tests/test_iqm_client.py

```diff
@@ -637,6 +641,137 @@ def test_get_quantum_architecture(
     unstub()
 
 
+def test_get_quality_metric_set(
+    sample_client, quality_metric_set_url, sample_quality_metric_set, quality_metric_set_success
+):
+    """Test retrieving the quality metric set."""
+    expect(requests, times=1).get(quality_metric_set_url, ...).thenReturn(quality_metric_set_success)
+
+    assert sample_client.get_quality_metric_set() == sample_quality_metric_set
+
+    verifyNoUnwantedInteractions()
+    unstub()
+
+
+def test_get_quality_metric_set_without_calset_id_does_not_cache(
+    sample_client, quality_metric_set_url, sample_quality_metric_set, quality_metric_set_success
+):
+    """
+    Tests that the correct quality metric set is returned in the case where default calset
+    changes between two invocations of get_quality_metric_set().
+    """
+    quality_metric_set_2 = {
+        'quality_metric_set_id': '3902d525-d8f4-42c0-9fa9-6bbd535b6c80',
+        'quality_metric_set_dut_label': 'M194_W0_P08_Z99',
+        'quality_metric_set_created_timestamp': '2024-02-10T08:57:04.605956',
+        'quality_metric_set_end_timestamp': '2024-02-10T08:57:04.605956',
+        'quality_metric_set_is_invalid': False,
+        'metrics': {
+            'QB1.t1_time': {
+                'value': '4.408139707188389e-05',
+                'unit': 's',
+                'uncertainty': '2.83049498694448e-06',
+                'timestamp': '2024-02-10T08:57:04.605956',
+            },
+            'QB1.t2_time': {
+                'value': '3.245501974471748e-05',
+                'unit': 's',
+                'uncertainty': '2.39049697699448e-06',
+                'timestamp': '2024-02-10T08:57:04.605956',
+            },
+        },
+    }
+
+    quality_metric_set_success_2 = MockJsonResponse(200, quality_metric_set_2)
+    expect(requests, times=2).get(quality_metric_set_url, ...).thenReturn(quality_metric_set_success).thenReturn(
+        quality_metric_set_success_2
+    )
+
+    assert sample_client.get_quality_metric_set() == sample_quality_metric_set
+    assert sample_client.get_quality_metric_set() == QualityMetricSet(**quality_metric_set_2)
```

Comment:
The idea is to compare to the reference value. The original instruction compares two 
`QualityMetricSet` objects constructed from the same data and if deserialization is broken, e.g. a field is missing, the test will not fail.

```suggestion
    assert sample_client.get_quality_metric_set().model_dump() == quality_metric_set_2
```